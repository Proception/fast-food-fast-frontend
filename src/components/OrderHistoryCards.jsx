import React, { Component } from 'react';

const OrderHistoryCards = ({ userOrders }) => (
    <div class="container">
        {userOrders.map(item => (
          <div class="old-order" key={item.order_id}>
            <p><label htmlFor="orderno">Order No. :</label><span>{item.order_id}</span></p>
            <p><label htmlFor="orderdate">Order Date :</label><span>{new Date(item.order_date).toGMTString()}</span></p>
            <p><label htmlFor="totalprice">Total Amount :</label><span>&#8358;{item.order_amount}</span></p>
            <p><label htmlFor="status">Status :</label> <span>{item.order_status}</span></p>
          </div>
        ))}
    </div>
  );
export default OrderHistoryCards;
