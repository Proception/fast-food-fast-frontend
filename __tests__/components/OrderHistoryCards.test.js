import React from 'react';
import { shallow } from 'enzyme';
import OrderHistoryCards from '../../src/components/OrderHistoryCards.jsx';
import props from '../../__mocks__/orderCardsMock.js';

let wrapper;
beforeEach(() => {
  const { orders } = props;
  wrapper = shallow(<OrderHistoryCards orders={orders}/>);
});
describe('OrderHistoryCards Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
