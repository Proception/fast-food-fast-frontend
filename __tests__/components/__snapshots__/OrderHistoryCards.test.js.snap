// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistoryCards Component should match snapshot 1`] = `
<div
  class="container"
>
  <div
    class="old-order"
  >
    <p>
      <label
        htmlFor="orderno"
      >
        Order No. :
      </label>
      <span />
    </p>
    <p>
      <label
        htmlFor="orderdate"
      >
        Order Date :
      </label>
      <span>
        Invalid Date
      </span>
    </p>
    <p>
      <label
        htmlFor="totalprice"
      >
        Total Amount :
      </label>
      <span>
        ₦
      </span>
    </p>
    <p>
      <label
        htmlFor="status"
      >
        Status :
      </label>
       
      <span />
    </p>
  </div>
  <div
    class="old-order"
  >
    <p>
      <label
        htmlFor="orderno"
      >
        Order No. :
      </label>
      <span />
    </p>
    <p>
      <label
        htmlFor="orderdate"
      >
        Order Date :
      </label>
      <span>
        Invalid Date
      </span>
    </p>
    <p>
      <label
        htmlFor="totalprice"
      >
        Total Amount :
      </label>
      <span>
        ₦
      </span>
    </p>
    <p>
      <label
        htmlFor="status"
      >
        Status :
      </label>
       
      <span />
    </p>
  </div>
  <div
    class="old-order"
  >
    <p>
      <label
        htmlFor="orderno"
      >
        Order No. :
      </label>
      <span />
    </p>
    <p>
      <label
        htmlFor="orderdate"
      >
        Order Date :
      </label>
      <span>
        Invalid Date
      </span>
    </p>
    <p>
      <label
        htmlFor="totalprice"
      >
        Total Amount :
      </label>
      <span>
        ₦
      </span>
    </p>
    <p>
      <label
        htmlFor="status"
      >
        Status :
      </label>
       
      <span />
    </p>
  </div>
  <div
    class="old-order"
  >
    <p>
      <label
        htmlFor="orderno"
      >
        Order No. :
      </label>
      <span />
    </p>
    <p>
      <label
        htmlFor="orderdate"
      >
        Order Date :
      </label>
      <span>
        Invalid Date
      </span>
    </p>
    <p>
      <label
        htmlFor="totalprice"
      >
        Total Amount :
      </label>
      <span>
        ₦
      </span>
    </p>
    <p>
      <label
        htmlFor="status"
      >
        Status :
      </label>
       
      <span />
    </p>
  </div>
</div>
`;
